dbt/dbt-run.py

```diff
@@ -133,10 +146,90 @@ def run_incremental_models():
   # update incremental models (and tables if there are any)
   subprocess.run(["dbt", "run",  "--profiles-dir", ".dbt", "--exclude", "config.materialized:view"])
 
+def get_pending_doc_count():
+  with connection() as conn:
+      with conn.cursor() as cur:
+          cur.execute(f"""
+              SELECT SUM(pending)
+              FROM {os.getenv('POSTGRES_SCHEMA')}.couchdb_progress
+          """)
+          return cur.fetchone()[0]
+
+def update_batch_status(timestamp, status):
+  with connection() as conn:
+    with conn.cursor() as cur:
+      # insert new entry
+      cur.execute(
+        f"INSERT INTO {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status (timestamp, status) VALUES (%s, %s);", [timestamp, status]
+      )
+      conn.commit()
+
+def get_last_processed_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.dbt_batch_status
+          WHERE status = 'success'
+      """)
+      result = cur.fetchone()
+      if result and result[0]:
+        return result[0]
+      return '1970-01-01 00:00:00'
+    
+def get_max_timestamp():
+  with connection() as conn:
+    with conn.cursor() as cur:
+      cur.execute(f"""
+          SELECT MAX(saved_timestamp)
+          FROM {os.getenv('POSTGRES_SCHEMA')}.document_metadata
+      """)
+      return cur.fetchone()[0]
+    
+def handle_batch_update_error(last_processed_timestamp, dataemon_interval):
+   print(f"Error running dbt for batch {last_processed_timestamp}")
+   update_batch_status(last_processed_timestamp, "error")
+   time.sleep(dataemon_interval)
+
+def handle_batch_update_success(last_processed_timestamp, dataemon_interval, max_timestamp):
+  update_batch_status(last_processed_timestamp, "success")
+
+  if max_timestamp == last_processed_timestamp:
+     time.sleep(dataemon_interval)
+   
+def run_dbt_in_batches():
+  last_processed_timestamp = get_last_processed_timestamp()
+  batch_size = int(os.getenv("DBT_BATCH_SIZE") or 10000)
+  dataemon_interval = int(os.getenv("DATAEMON_INTERVAL") or 5)
+
+  while True:
+     update_dbt_deps()
+     result = subprocess.run([
+        "dbt", "run",
+        "--profiles-dir", ".dbt",
+        "--vars", f'{{start_timestamp: "{last_processed_timestamp}", batch_size: {batch_size}}}',
+        "--exclude", "config.materialized:view"
+      ])
+     
+     if result.returncode != 0:
+        handle_batch_update_error(last_processed_timestamp, dataemon_interval)
+        continue
```

Comment:
I believe we should add some error handling here. Does the command actually log the error? I'm not sure how valuable it is to record the failure without storing any information about _why_ it failed.

Reply:
So it is actually logged somewhere? I don't understand.

Reply:
```
2024-10-08 13:54:44 cht_pipeline_base.contacts.relationships_person_uuid__uuid__ref_contact_
2024-10-08 13:54:44 cht_pipeline_base.contacts.relationships_place_uuid__uuid__ref_contact_
2024-10-08 13:54:44 cht_pipeline_base.users.relationships_user_contact_uuid__uuid__ref_contact_
2024-10-08 13:54:44 cht_pipeline_base.users.relationships_user_user_id__uuid__ref_contact_
2024-10-08 13:54:44 cht_pipeline_base.contacts.unique_contact_uuid
2024-10-08 13:54:44 cht_pipeline_base.contacts.unique_person_uuid
2024-10-08 13:54:44 cht_pipeline_base.sqltest.user
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_contact_model_transformation_and_data_integrity
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_contact_types
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_data_record_model_transformation_and_data_integrity
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_document_metadata
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_patient_model_transformation_and_data_integrity
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_person_model_transformation_and_data_integrity
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_place_model_transformation_and_data_integrity
2024-10-08 13:54:44 unit_test:cht_pipeline_base.test_user_model_transformation_and_data_integrity
2024-10-08 13:54:50 Usage: dbt run [OPTIONS]
2024-10-08 13:54:50 Try 'dbt run -h' for help.
2024-10-08 13:54:50 
2024-10-08 13:54:50 Error: Invalid value for '--profiles-dir': Path '.dbt1' does not exist.
2024-10-08 13:54:50 Error running dbt for batch 2024-10-07 14:29:28.198000
```
Example container logs for when a dbt run fails.